src/main.rs

```diff
@@ -0,0 +1,182 @@
+                    if let Concept::Attribute(Attribute { value: Value::String(value), .. }) = concept {
+                        println!("{}",value);
+                    }
+                }
+            }
+            Err(err) => {
+                panic!("An error occurred fetching answers of a Match query: {err}")
+            }
+        }
+    }
+
+    Ok(())
+}
+
+async fn query3(connection: Connection)->std::io::Result<()>{    
+    let databases = DatabaseManager::new(connection.clone());
+    let session = Session::new(databases.get(TEST_DATABASE).await.unwrap(), Data).await.unwrap();
+    let transaction = session.transaction(Read).await.unwrap();
+    println!("Query chosen : Get problem-name of problems with a particular tag with rating >= x");
+    println!("Choose rating x");
+    let x: String=read!("{}\n");
+    println!("Choose tag");
+    let tag: String=read!();
+    let query_part_1 ="match $p isa problem, has problem-name $m, has rating >= ";
+    let query_part_2 = "; $y isa topic, has topic-name \"";
+    let query_part_3 = "\"; $tt ($p, $y) isa possesses-tag; get $m;";
+    let mut full_query = "".to_owned();
+    full_query = full_query + query_part_1 + x.as_str() + query_part_2 + tag.as_str() + query_part_3;
+
+    println!("{}", full_query);
+
+    let mut answer_stream = transaction.query().match_(&full_query.as_str()).unwrap();
+    while let Some(result) = answer_stream.next().await{
+        match result {
+            Ok(concept_map) => {
+                for (_, concept) in concept_map {
+                    if let Concept::Attribute(Attribute { value: Value::String(value), .. }) = concept {
+                        println!("{}",value);
+                    }
+                }
+            }
+            Err(err) => {
+                panic!("An error occurred fetching answers of a Match query: {err}")
+            }
+        }
+    }
+
+    Ok(())
+}
+
+async fn run_io(connection: Connection){
+    println!("Welcome to the codeforces data model project. Please choose what kind of query you would like to make from the following options");
+    println!("Please select your entry using the number of the query");
+    println!("Options");
+    println!("1) Get names of all coders with rating >= x");
+    println!("2) Get IDs of problems with a particular tag");
+    println!("3) Get problem-name of problems with a particular tag with rating >= x");
+    let qtype: i32=read!();
+    if qtype == 1{
+        let _ = query1(connection).await;
+    }else if qtype == 2{
+        let _ = query2(connection).await;
+    }else if qtype == 3{
+        let _ = query3(connection).await;
+    }else{
+        println!("Retry, invalid query option chosen");
+    }
+}
+
+async fn mymain()->std::io::Result<()>{
+
+    let con=new_core_connection().expect("Line: 74");
+    load_schema(con.clone()).await?;
+    run_io(con.clone()).await;
+
+    Ok(())
+}
+
+#[tokio::main]
+async fn main(){
```

Comment:
`main` can return a `Result`. So you don't need to create a separate `mymain` function.

Reply:
Fixed!